spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigUtils.java

```diff
@@ -77,6 +81,27 @@ static String getApplicationNamespace(String namespace, String configurationTarg
 		throw new NamespaceResolutionFailedException("unresolved namespace");
 	}
 
+	/*
+	 * namespace that reaches this point is absolutely present, otherwise this would have
+	 * resulted in a NamespaceResolutionFailedException
+	 */
+	static Map<String, String> getConfigMapData(CoreV1Api client, String namespace, String name) {
+		V1ConfigMap configMap;
+		try {
+			configMap = client.readNamespacedConfigMap(name, namespace, null, null, null);
+		}
+		catch (ApiException e) {
```

Comment:
this emulates what fabric8 is doing. Fabric8 returns `null` when something is missing, k8s-native a 404